source/base/Text.ooc

```diff
@@ -135,10 +135,15 @@ Text: cover {
 		this _buffer free(criteria)
 	}
 	slice: func (start, distance: Int) -> This {
-		result := This new(this _buffer slice(start, distance))
-		if (this _buffer owner == Owner Receiver)
-			result = result copy() // TODO: Could we be smarter here?
-		this free(Owner Receiver)
+		result: This
+		if (start == 0 && distance == this count)
+			result = this
+		else {
+			result = This new(this _buffer slice(start, distance))
+			if (this _buffer owner == Owner Receiver)
+				result = result copy() // TODO: Could we be smarter here?
+			this free(Owner Receiver)
+		}
 		result
 	}
```

Comment:
I would say that Thomas implementation is a bit easier to read but I don't dislike your somewhat shorter solution either. The question I ask is why at all do we need this special case at all?